addons/payment/models/payment_acquirer.py

```diff
@@ -45,119 +42,116 @@ class PaymentAcquirer(osv.Model):
-        'company_id': fields.many2one('res.company', 'Company', required=True),
-        'pre_msg': fields.html('Help Message', translate=True,
-                               help='Message displayed to explain and help the payment process.'),
-        'post_msg': fields.html('Thanks Message', translate=True,
-                                help='Message displayed after having done the payment process.'),
-        'view_template_id': fields.many2one('ir.ui.view', 'Form Button Template', required=True),
-        'registration_view_template_id': fields.many2one('ir.ui.view', 'S2S Form Template',
-                                                         domain=[('type', '=', 'qweb')],
-                                                         help="Template for method registration"),
-        'environment': fields.selection(
-            [('test', 'Test'), ('prod', 'Production')],
-            string='Environment', required=True, oldname='env'),
-        'website_published': fields.boolean(
-            'Visible in Portal / Website', copy=False,
-            help="Make this payment acquirer available (Customer invoices, etc.)"),
-        'auto_confirm': fields.selection(
-            [('none', 'No automatic confirmation'),
-             ('at_pay_confirm', 'At payment with acquirer confirmation'),
-             ('at_pay_now', 'At payment no acquirer confirmation needed')],
-            string='Order Confirmation', required=True),
-        'pending_msg': fields.html('Pending Message', translate=True, help='Message displayed, if order is in pending state after having done the payment process.'),
-        'done_msg': fields.html('Done Message', translate=True, help='Message displayed, if order is done successfully after having done the payment process.'),
-        'cancel_msg': fields.html('Cancel Message', translate=True, help='Message displayed, if order is cancel during the payment process.'),
-        'error_msg': fields.html('Error Message', translate=True, help='Message displayed, if error is occur during the payment process.'),
-        # Fees
-        'fees_active': fields.boolean('Add Extra Fees'),
-        'fees_dom_fixed': fields.float('Fixed domestic fees'),
-        'fees_dom_var': fields.float('Variable domestic fees (in percents)'),
-        'fees_int_fixed': fields.float('Fixed international fees'),
-        'fees_int_var': fields.float('Variable international fees (in percents)'),
-        'sequence': fields.integer('Sequence', help="Determine the display order"),
-        'module_id': fields.many2one('ir.module.module', string='Corresponding Module'),
-        'module_state': fields.related('module_id', 'state', type='char', string='Installation State'),
-        'description': fields.html('Description'),
-    }
-
-    image = openerp.fields.Binary("Image", attachment=True,
+    _provider_selection = lambda self: self._get_providers()
```

Comment:
Unnecessary indirection, the method can be provided by name to the field.

If all payments were migrated simultaneously, even that would be unnecessary: other methods could use `selection_add` to extend the `provider` field.